Podverse/ClipsTableViewController.swift

```diff
@@ -15,66 +15,59 @@ class ClipsTableViewController: PVViewController {
     
-    var filterTypeSelected: ClipFilterType = .allPodcasts {
+    var filterTypeSelected: ClipFilter = .allPodcasts {
         didSet {
-            self.filterType.setTitle(filterTypeSelected.text + "\u{2304}", for: .normal)
+            self.tableViewHeader.filterTitle = self.filterTypeSelected.text
             UserDefaults.standard.set(filterTypeSelected.text, forKey: kClipsTableFilterType)
         }
     }
-
+    var sortingTypeSelected: ClipSorting = .topWeek {
+        didSet {
+            self.tableViewHeader.sortingTitle = sortingTypeSelected.text
+            UserDefaults.standard.set(sortingTypeSelected.text, forKey: kClipsTableSortingType)
+        }
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         activityIndicator.hidesWhenStopped = true
         showIndicator()
         
         self.clipQueryActivityIndicator.hidesWhenStopped = true
         self.clipQueryMessage.isHidden = true
         
-        if let savedFilterType = UserDefaults.standard.value(forKey: kClipsTableFilterType) as? String, let clipFilterType = ClipFilterType(rawValue: savedFilterType) {
+        if let savedFilterType = UserDefaults.standard.value(forKey: kClipsTableFilterType) as? String, let clipFilterType = ClipFilter(rawValue: savedFilterType) {
             self.filterTypeSelected = clipFilterType
         } else {
             self.filterTypeSelected = .allPodcasts
-            
         }
         
-        retrieveClips()
-    }
-    
-    @IBAction func updateFilter(_ sender: Any) {
-        let alert = UIAlertController(title: "Clips From", message: nil, preferredStyle: .alert)
-        
-        alert.addAction(UIAlertAction(title: "Subscribed", style: .default, handler: { action in
-            self.resetClipQuery()
-            self.filterTypeSelected = .subscribed
-            self.retrieveClips()
-        }))
-        
-        alert.addAction(UIAlertAction(title: "All Podcasts", style: .default, handler: { action in
-            self.resetClipQuery()
-            self.filterTypeSelected = .allPodcasts
-            self.retrieveClips()
-        }))
-        
-        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+        if let savedSortingType = UserDefaults.standard.value(forKey: kClipsTableSortingType) as? String, let clipSortingType = ClipSorting(rawValue: savedSortingType) {
+            self.sortingTypeSelected = clipSortingType
+        } else {
+            self.sortingTypeSelected = .topWeek
+        }
         
-        self.present(alert, animated: true, completion: nil)
+        retrieveClips()
     }
     
-    
-    @IBAction func updateSorting(_ sender: Any) {
-        
+    override func viewWillAppear(_ animated: Bool) {
+        self.tableViewHeader.filterTitle = self.filterTypeSelected.text
+        self.tableViewHeader.sortingTitle = self.sortingTypeSelected.text
+        self.tableViewHeader.delegate = self
```

Comment:
You might want to add this and the next line into viewDidLoad so you don't setup the views and set the delegate every time the view appears.

Reply:
Ok done